src/coreclr/jit/codegenarm.cpp

```diff
@@ -653,6 +658,254 @@ void CodeGen::genJumpTable(GenTree* treeNode)
     genProduceReg(treeNode);
 }
 
+//------------------------------------------------------------------------
+// genLockedInstructions: Generate code for a GT_XADD or GT_XCHG node.
+//
+// Arguments:
+//    treeNode - the GT_XADD/XCHG node
+//
+void CodeGen::genLockedInstructions(GenTreeOp* treeNode)
```

Comment:
The `genLocked*` methods added here looks similar to the one in arm64. can the logic be shared and have just a single method in `codegenarmarch.cpp`?